src/main/java/org/threadly/concurrent/future/Watchdog.java

```diff
@@ -0,0 +1,195 @@
+    this(getStaticScheduler(), timeoutInMillis, sendInterruptToTrackedThreads);
+  }
+  
+  /**
+   * Constructs a new {@link Watchdog} with a scheduler of your choosing.  It is critical that 
+   * this scheduler has a free thread available to inspect futures which may not have completed in 
+   * the given timeout.  You may want to use a limiter to ensure that there are threads available.
+   * 
+   * @param scheduler Scheduler to schedule task to look for expired futures
+   * @param timeoutInMillis Time in milliseconds that futures will be set to error if they are not done
+   * @param sendInterruptToTrackedThreads If {@code true}, and a thread is provided with the future, 
+   *                                        an interrupt will be sent on timeout
+   */
+  public Watchdog(SimpleSchedulerInterface scheduler, long timeoutInMillis, 
+                  boolean sendInterruptToTrackedThreads) {
+    this.scheduler = scheduler;
+    this.timeoutInMillis = timeoutInMillis;
+    this.sendInterruptToTrackedThreads = sendInterruptToTrackedThreads;
+    this.checkRunner = new CheckRunner();
+    this.futures = new ConcurrentLinkedQueue<FutureWrapper>();
+    this.checkRunnerStatus = new AtomicInteger(-1);
+  }
+  
+  /**
+   * Watch a given {@link ListenableFuture} to ensure that it completes within the constructed 
+   * time limit.  If the future is not marked as done by the time limit then it will be 
+   * completed by invoking {@link ListenableFuture#cancel(boolean)} with {@code true} to interrupt 
+   * the running thread.
+   * 
+   * @param future Future to inspect to ensure completion
+   */
+  public void watch(ListenableFuture<?> future) {
+    if (future == null || future.isDone()) {
+      return;
+    }
+    
+    final FutureWrapper fw = new FutureWrapper(future);
+    futures.add(fw);
+    // we attempt to remove the future on completion to reduce inspection needed
+    future.addListener(new Runnable() {
+      @Override
+      public void run() {
+        futures.remove(fw);
+      }
+    });
+    
+    while (true) {
+      if (checkRunnerStatus.get() == -1) {
+        if (checkRunnerStatus.compareAndSet(-1, 0)) {
+          scheduler.schedule(checkRunner, timeoutInMillis);
+          return;
+        }
+      } else if (checkRunnerStatus.get() == 1) {
+        if (checkRunnerStatus.compareAndSet(1, 2)) {
+          return;
+        }
+      } else {
+        // either already scheduled, or already marked as more added
+        return;
+      }
+    }
+  }
+
+  /**
+   * <p>Just a simple wrapper class so we can hold not just the future, but what time the future 
+   * will expire at.</p>
+   * 
+   * @author jent - Mike Jensen
+   * @since 4.0.0
+   */
+  private class FutureWrapper {
+    public final long expireTime;
+    private final ListenableFuture<?> future;
+
+    public FutureWrapper(ListenableFuture<?> future) {
+      this.expireTime = Clock.accurateForwardProgressingMillis() + timeoutInMillis;
```

Comment:
Yea, I went back and forth too.  I could be convinced either way really.  It was originally lastKnown, but I decided it's safer to just do accurate.